### PR TITLE
cpu: aarch64: acl: disallow large padding in ACL wino to be consist with oneDNN

### PR DESCRIPTION
**Pull request by diaena**
# Description

This PR fixes a recently identified issue (https://github.com/oneapi-src/oneDNN/issues/1421#issuecomment-1185589357)
in the Aarch64 build with Compute Library. A recent update to the winograd
kernels in ACL have added support for arbitrary padding when oneDNN expects this
case to throw an exception in the reference kernels.

To reproduce: build oneDNN with ACL off the main branch from MLPlatform.org and
run the `test_iface_wino_convolution` test.

This patch disallows large padding configurations in the ACL winograd
integration to make it consistent with the oneDNN library.

# Checklist

## General

- [X] Do all unit and benchdnn tests (`make test` and `make test_benchdnn_*`) pass locally for each commit?
- [X] Have you formatted the code using clang-format?

## Performance improvements

- [N/A ] Have you submitted performance data that demonstrates performance improvements?

### New features

- [N/A] Have you published an RFC for the new feature?
- [N/A] Was the RFC approved?
- [N/A] Have you added relevant tests?

### Bug fixes

- [X] Have you included information on how to reproduce the issue (either in a github issue or in this PR)?
- [N/A] Have you added relevant regression tests?

## [RFC](https://github.com/oneapi-src/oneDNN/tree/rfcs) PR

- [ ] Does RFC document follow the [template](https://github.com/oneapi-src/oneDNN/blob/rfcs/rfcs/template.md#onednn-design-document-rfc)?
- [ ] Have you added a link to the rendered document?

### CODE REVIEW

**Review by dzarukin — APPROVED** [review]
LGTM. By the way, has the path to modify the test been considered? All in all, if Winograd brings more performance with ACL over direct approach in cases when padding is larger, it may be better to keep it enabled. This is just a thought.

**Comment by nSircombe** [comment]
Hi @dzarukin,
In this case the priority is to have the fix in place ahead of the ACL 22.08 release, which will expose the test failure. We haven't looked in any detail at the potential benefits of leaving this functionality in place and modifying the tests, although we did consider it.

Generally, we find the cases where this Winograd impl. can be beneficial up are far from widespread, and it doesn't provide bf16 support like the GEMM kernels, added to this we're also working on a number of changes in ACL which may make the Winograd kernel even less relevant.
In short we're not confident that we will keep this ACL kernel in the mix in the long term, so a 'fixup' looks like the more pragmatic approach for now at least!

**Comment by dzarukin** [comment]
> Hi @dzarukin, In this case the priority is to have the fix in place ahead of the ACL 22.08 release, which will expose the test failure. We haven't looked in any detail at the potential benefits of leaving this functionality in place and modifying the tests, although we did consider it.
> 
> Generally, we find the cases where this Winograd impl. can be beneficial up are far from widespread, and it doesn't provide bf16 support like the GEMM kernels, added to this we're also working on a number of changes in ACL which may make the Winograd kernel even less relevant. In short we're not confident that we will keep this ACL kernel in the mix in the long term, so a 'fixup' looks like the more pragmatic approach for now at least!

Hi @nSircombe, sounds good to me, thank you for the explanation.

**Comment by dzarukin** [comment]
Thank you for the contribution!
